File: result/translate/chatgpt/java2python/python/NUMBER_VISIBLE_BOXES_PUTTING_ONE_INSIDE_ANOTHER.py
def f_gold(arr, n):
    from queue import Queue
    from collections import deque
    
    q = deque()
    arr = arr[:n]
    arr.sort()
    q.append(arr[0])
    
    for i in range(1, n):
        now = q[0]
        if arr[i] >= 2 * now:
            q.popleft()
        q.append(arr[i])
    
    return len(q)

File: result/translate/chatgpt/python2java/java/NUMBER_VISIBLE_BOXES_PUTTING_ONE_INSIDE_ANOTHER.java
public static int f_gold(int[] arr, int n) {
    Deque<Integer> q = new LinkedList<>();
    arr = Arrays.copyOf(arr, n);
    Arrays.sort(arr);
    q.add(arr[0]);
    for (int i = 1; i < n; i++) {
        int now = q.peekFirst();
        if (arr[i] >= 2 * now) {
            q.pollFirst();
        }
        q.add(arr[i]);
    }
    return q.size();
}